core/bourbon/settings/_bourbon-get-setting.scss

```diff
@@ -0,0 +1,13 @@
+/// Return a Bourbon setting.
+///
+/// @argument {string} $setting
+///
+/// @example scss
+///   _bourbon-get-setting(rails-asset-pipeline)
+///
+/// @access private
+
+@function _bourbon-get-setting($setting) {
+  $settings: map-merge($_bourbon-defaults, $bourbon);
```

Comment:
Interesting point. We were discussing performance around this…

I like the clarity of your first option and am going to run with that for the moment, but will try to benchmark sometime.